### ISSUE TITLE
Isolate individual tests and parallel test runs

### ISSUE THREAD

**Issue by katspaugh** [gnosis/safe-react-e2e-tests#3]
Make sure the tests are self-sufficient and encapsulated, not affecting each other. Also when several people/pipelines are running the tests in parallel, these runs should also not affect each other.

Clear/restore the shared state after each `it` or `describe`.

E.g. if funds are sent, they should be returned back.

**Stretch goal:**
Create a pool of 100 safes with preloaded owners and funds, and pick one for each suite run. E.g. pick a safe based on the timestamp % 100.
